src/test/java/org/openrewrite/java/logging/log4j/ConvertJulEnteringTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging.log4j;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.java.ChangeType;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class ConvertJulEnteringTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipes(new ConvertJulEntering(),
+            new ChangeType("java.util.logging.Logger", "org.apache.logging.log4j.Logger", true))
+          .parser(JavaParser.fromJavaVersion().classpath("log4j-api"));
+    }
+
+    @Test
+    @DocumentExample
+    void enteringToTraceEntry() {
+        rewriteRun(
+          // language=java
+          java(
+            """
+              import java.util.logging.Logger;
+
+              class Test {
+                  void method(Logger logger) {
+                    logger.entering("Test", "method");
+                    logger.entering("Test", "method", "param");
+                    logger.entering("Test", "method", new Object[]{"param1", "param2"});
+                  }
+              }
+              """,
+            """
+              import org.apache.logging.log4j.Logger;
+
+              class Test {
+                  void method(Logger logger) {
+                    logger.traceEntry(null);
+                    logger.traceEntry(null, "param");
+                    logger.traceEntry(null, new Object[]{"param1", "param2"});
```

Comment:
I might just lack some context, but why pass in a `null` first argument here?

Reply:
Specifically for the first of the three statements I'm wondering; the others maybe just as well?

Reply:
Nice catch, the first one can be `logger.traceEntry()`.

The remaining two are a little hack: since there is no `traceEntry(Object...)` method, I am calling [`traceEntry(String, Object...)`](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger#traceEntry(java.lang.String,java.lang.Object...)). The default implementation of `FlowMessageFactory` (cf. [source code](https://github.com/apache/logging-log4j2/blob/2d3ab5268a3ff7dbdc127eb480086a708b10acf6/log4j-api/src/main/java/org/apache/logging/log4j/message/DefaultFlowMessageFactory.java#L194)) generates an appropriate format string if the format parameter is empty or `null`.

Reply:
`logger.traceEntry(null)` will have the same effect, but `logger.traceEntry()` seems more elegant.

Reply:
Fixed in https://github.com/openrewrite/rewrite-logging-frameworks/commit/a69a8b85c43e3f71b8fdd0a722bdc130e55b6b25 thanks!